File: code/python/61.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: ListNode, k: int) -> ListNode:
        if head is None or head.next is None: return head

        length = 0
        p = head
        while p is not None:
            length += 1
            p = p.next

        k = k % length
        if k == 0: return head

        left = length - k
        pre = head
        cur = head
        c = 0
        while c < left:
            pre = cur
            cur = cur.next
            c += 1
        pre.next = None
        res = cur
        while cur.next is not None:
            cur = cur.next
        cur.next = head
        return res

File: code/java/61.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head==null)
            return null;
        ListNode result;
        int len = 0;
        ListNode s = head;
        while(s!=null){
            len++;
            s = s.next;
        }
        k = k%len;
        if(k==0)
            return head;
        else{
            int stop = len-k;
            int i = 1;
            s = head;
            while(i<stop){
                s = s.next;
                i++;
            }
            result = s.next;
            s.next = null;
            s = result;
            while(s.next!=null)
                s = s.next;
            s.next = head;
            return result;
        }
    }
}